src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -461,14 +461,23 @@ private int ComputeOffsetToMetadata(int ilStreamLength)
             return OffsetToILStream + BitArithmeticUtilities.Align(ilStreamLength, 4);
         }
 
-        private const int ImageDebugDirectoryBaseSize =
-            sizeof(uint) +   // Characteristics
-            sizeof(uint) +   // TimeDataStamp
-            sizeof(uint) +   // Version
-            sizeof(uint) +   // Type
-            sizeof(uint) +   // SizeOfData
-            sizeof(uint) +   // AddressOfRawData
-            sizeof(uint);    // PointerToRawData
+        /// <summary>
+        /// The size of a single entry in the "Debug Directory (Image Only)"
+        /// </summary>
+        private const int ImageDebugDirectoryEntrySize =
+                    sizeof(uint) +   // Characteristics
+                    sizeof(uint) +   // TimeDataStamp
+                    sizeof(uint) +   // Version
+                    sizeof(uint) +   // Type
+                    sizeof(uint) +   // SizeOfData
+                    sizeof(uint) +   // AddressOfRawData
+                    sizeof(uint);    // PointerToRawData
+
+        /// <summary>
+        /// The size of our debug directory: one entry for debug information, and an optional second one indicating
+        /// that the timestamp is deterministic (i.e. not really a timestamp)
+        /// </summary>
+        private int ImageDebugDirectoryBaseSize => (_deterministic ? 2 : 1) * ImageDebugDirectoryEntrySize;
```

Comment:
This isn't right. We now need to produce the debug table when debug is enabled (which adds one entry plus some data) or when determinism is enabled (which adds one entry). Fixing this will require changes in a few places.